src/geo_types_to_wkt.rs

```diff
@@ -0,0 +1,441 @@
+/// ```
+impl<T> ToWkt<T> for geo_types::MultiLineString<T>
+where
+    T: CoordFloat + std::fmt::Display,
+{
+    fn to_wkt(&self) -> Wkt<T> {
+        Wkt {
+            item: g_mline_to_w_mline(self).as_item(),
+        }
+    }
+}
+
+/// # Examples
+/// ```
+/// use geo_types::{polygon, Polygon, MultiPolygon};
+/// use wkt::ToWkt;
+///
+/// // triangle
+/// let polygon_1: Polygon<f64> = polygon![(x: 0., y: 0.), (x: 4., y: 0.), (x: 2., y: 4.), (x: 0., y: 0.)];
+/// // square
+/// let polygon_2: Polygon<f64> = polygon![(x: 4., y: 4.), (x: 8., y: 4.), (x: 8., y: 8.), (x: 4., y: 8.), (x: 4., y: 4.)];
+/// let multi_polygon: MultiPolygon<f64> = MultiPolygon::new(vec![polygon_1, polygon_2]);
+///
+/// assert_eq!(multi_polygon.wkt_string(), "MULTIPOLYGON(((0 0,4 0,2 4,0 0)),((4 4,8 4,8 8,4 8,4 4)))");
+/// ```
+impl<T> ToWkt<T> for geo_types::MultiPolygon<T>
+where
+    T: CoordFloat + std::fmt::Display,
+{
+    fn to_wkt(&self) -> Wkt<T> {
+        Wkt {
+            item: g_mpolygon_to_w_mpolygon(self).as_item(),
+        }
+    }
+}
+
+/// # Examples
+/// ```
+/// use geo_types::{line_string, LineString, polygon, Polygon, GeometryCollection};
+/// use wkt::ToWkt;
+///
+/// let polygon: Polygon<f64> = polygon![(x: 0., y: 0.), (x: 4., y: 0.), (x: 2., y: 4.), (x: 0., y: 0.)];
+/// let line_string: LineString<f64> = line_string![(x: 1., y: 2.), (x: 3., y: 4.), (x: 5., y: 6.)];
+/// let geometry_collection: GeometryCollection<f64> = GeometryCollection::new_from(vec![polygon.into(), line_string.into()]);
+///
+/// assert_eq!(geometry_collection.wkt_string(), "GEOMETRYCOLLECTION(POLYGON((0 0,4 0,2 4,0 0)),LINESTRING(1 2,3 4,5 6))");
+/// ```
+impl<T> ToWkt<T> for geo_types::GeometryCollection<T>
+where
+    T: CoordFloat + std::fmt::Display,
+{
+    fn to_wkt(&self) -> Wkt<T> {
+        Wkt {
+            item: g_geocol_to_w_geocol(self).as_item(),
+        }
+    }
+}
+
+/// # Examples
+/// ```
+/// use geo_types::{coord, Rect};
+/// use wkt::ToWkt;
+///
+/// let rect: Rect<f64> = Rect::new(coord!(x: 4., y: 4.), coord!(x: 8., y: 8.));
+///
+/// assert_eq!(rect.wkt_string(), "POLYGON((4 4,4 8,8 8,8 4,4 4))");
+/// ```
+impl<T> ToWkt<T> for geo_types::Rect<T>
+where
+    T: CoordFloat + std::fmt::Display,
+{
+    fn to_wkt(&self) -> Wkt<T> {
+        Wkt {
+            item: g_rect_to_w_polygon(self).as_item(),
+        }
+    }
+}
+
+/// # Examples
+/// ```
+/// use geo_types::{coord, Triangle};
+/// use wkt::ToWkt;
+///
+/// let triangle: Triangle<f64> = Triangle::new(coord!(x: 0., y: 0.), coord!(x: 4., y: 0.), coord!(x: 2., y: 4.));
+///
+/// assert_eq!(triangle.wkt_string(), "POLYGON((0 0,4 0,2 4,0 0))");
+/// ```
+impl<T> ToWkt<T> for geo_types::Triangle<T>
+where
+    T: CoordFloat + std::fmt::Display,
+{
+    fn to_wkt(&self) -> Wkt<T> {
+        Wkt {
+            item: g_triangle_to_w_polygon(self).as_item(),
+        }
+    }
+}
+
+fn g_point_to_w_coord<T>(g_point: &geo_types::Coordinate<T>) -> Coord<T>
```

Comment:
Everything after this point is the geo-types specific code cut/pasted out of `towkt.rs` which is no longer tied to geo-types.